exp/api/v1beta2/awsmachinepool_types.go

```diff
@@ -83,6 +83,10 @@ type AWSMachinePoolSpec struct {
 	// Enable or disable the capacity rebalance autoscaling group feature
 	// +optional
 	CapacityRebalance bool `json:"capacityRebalance,omitempty"`
+
+	// SuspendProcesses defines a list of processes to suspend for the given ASG. This is constantly reconciled.
+	// If a process is removed from this list it will automatically be resumed.
+	SuspendProcesses []string `json:"suspendProcesses,omitempty"`
```

Comment:
I wonder if we could give a better user experience here?  So instead of the users having to the know the right strings could we do something like this for example:

```yaml
suspendProcesses:
  processes:
     azrebalance: true
     launch: false
     .....
```

And the all case:

```yaml
suspendProcesses:
   all: true
```

We did something similar with the eks logging.

Reply:
Interesting. Sounds good! Because people who load in the definition can actually autocomplete these values. That's a good idea, I like that. :) Thanks!

Reply:
It does....it started me thinking :)

Reply:
Awesome! :) I resolved your previous comment. :) Thanks